automation/src/test/scala/org/broadinstitute/dsde/test/api/WorkspaceApiSpec.scala

```diff
@@ -338,6 +339,72 @@ class WorkspaceApiSpec extends TestKit(ActorSystem("MySpec")) with FreeSpecLike
         }
       }
     }
+
+    "should allow writers" - {
+      "to launch workflows if they have can-compute permission" in {
+        withCleanBillingProject(owner) { projectName =>
+          withWorkspace(projectName, prependUUID("writer-can-launch-workflow"), aclEntries = List(AclEntry(studentA.email, WorkspaceAccessLevel.Writer, canCompute = Some(true)))) { workspaceName =>
+            Rawls.entities.importMetaData(projectName, workspaceName, entity)(ownerAuthToken)
+
+            withMethod("writer-method", MethodData.SimpleMethod) { methodName =>
+              val method = MethodData.SimpleMethod.copy(methodName = methodName)
+
+              Rawls.methodConfigs.createMethodConfigInWorkspace(projectName, workspaceName,
+                method, method.methodNamespace, method.methodName, 1,
+                SimpleMethodConfig.inputs, SimpleMethodConfig.outputs, method.rootEntityType)(ownerAuthToken)
+
+              Orchestration.methods.setMethodPermissions(
+                method.methodNamespace,
+                method.methodName,
+                method.snapshotId,
+                studentA.email,
+                "OWNER"
+              )(ownerAuthToken)
+
+              val submissionId = Rawls.submissions.launchWorkflow(projectName, workspaceName, method.methodNamespace, method.methodName, method.rootEntityType, "participant1", "this", false)(studentAToken)
+              // make sure the submission has not errored out
+              eventually {
+                val submissionStatus = Rawls.submissions.getSubmissionStatus(projectName, workspaceName, submissionId)(studentAToken)._1
+                List("Accepted", "Evaluating", "Submitting", "Submitted") should contain (submissionStatus)
+              }
+            }(ownerAuthToken)
+          }(ownerAuthToken)
+        }
+      }
+    }
+
+    "should not allow writers" - {
+      "to launch workflows if they don't have can-compute permission" in {
+        withCleanBillingProject(owner) { projectName =>
+          withWorkspace(projectName, prependUUID("writer-cannot-launch-workflow"), aclEntries = List(AclEntry(studentA.email, WorkspaceAccessLevel.Writer, canCompute = Some(false)))) { workspaceName =>
+            Rawls.entities.importMetaData(projectName, workspaceName, entity)(ownerAuthToken)
+
+            withMethod("writer-method", MethodData.SimpleMethod) { methodName =>
```

Comment:
Question, because I'm not sure: Does having 2 methods with the same name ("writer-method") cause issues? You're using the same name in the previous test as well. If not, then cool :)

Reply:
it doesn't matter because this spec is not `with ParallelTestExecution`. tests in same spec run sequentially